js/flightlog_parser.js

```diff
@@ -582,23 +582,20 @@ var FlightLogParser = function(logData) {
                 // Extract the firmware revision in case of Betaflight/Raceflight/Cleanfligh 2.x/Other
                 var matches = fieldValue.match(/(.*flight).* (\d+)\.(\d+)(\.(\d+))*/i);
                 if(matches!=null) {
-                
-                    // The that.sysConfig.firmwareType usually has been detected before that. Default value.
-                    if (typeof that.sysConfig.firmwareType === 'undefined' || !that.sysConfig.firmwareType) {
-                	    that.sysConfig.firmwareType = FIRMWARE_TYPE_BETAFLIGHT;                    
+
+                    // Detecting Betaflight requires looking at the revision string
+                    if (matches[1] === "Betaflight") {
```

Comment:
I coded this the way I did as NOWHERE else does the firmwareType get set to FIRMWARE_TYPE_BETAFLIGHT, so the only case that will hit your second if would be if the preceeding if is true. Additionally every other case of setting the firmwareType is accompanied with toggling the isBaseF/isCF/isBF/isINAV classes on the HTML element, so the way it is currently coded in my pull request is more consistent and just as functional.